<a href="https://colab.research.google.com/github/gilesitorr/Estadistica_Avanzada/blob/main/Parcial_2/Tarea_4/EA_parcial2_tarea4.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# __Estadística Avanzada__
# Parcial 2 - Tarea 4
## _Gil Estéfano Rodríguez Rivera_

Todas las bases de datos utilizadas así como el desarrollo se encuentra en _GitHub_ en https://github.com/gilesitorr/Estadistica_Avanzada/tree/main/Parcial_2/Tarea_4

__Nota:__ En este Notebook se usó el kernel de R, aunque la extensión sea ipynb.

In [ ]:
# Antes de comenzar, se importan las librerías a usar
library(dplyr)

## Biostatistics: A Foundation for Analysis in the Health Sciences
### Wayne W. Daniel & Chad L. Cross

### 12.3.1


__The following table shows the distribution of uric acid determinations taken on 250 patients. Test the goodness-of-fit of these data to a normal distribution with $\mu=5.74$ and $\sigma=2.01$. Let $\alpha=0.01$.__

< 1, 1 

1 to 1.99, 5 

2 to 2.99, 15 

3 to 3.99, 24 

4 to 4.99, 43 

5 to 5.99, 50

6 to 6.99, 45

7 to 7.99, 30

8 to 8.99, 22

9 to 9.99, 10

10 or higher, 5

In [50]:
# Los parámetros de la distribución propuesta
mu <- 5.74
sigma <- 2.01

# Los datos

# Las cotas inferiores de los grupos (usa -1000 en lugar de -infinito porque
# en la práctica es equivalente)
inferior <- c(-1000, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10)
# Las cotas superiores de los grupos (usa 1000 en lugar de infinito porque
# en la práctica y en este contexto es equivalente)
superior <- c(1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 1000)
# Las frecuencias de cada grupo
values <- c(1, 5, 15, 24, 43, 50, 45, 30, 22, 10, 5)

# El tamaño de la muestra
numero <- sum(values)

# Se ajustan las frecuencias esperadas
esperadas <- numero*(pnorm((superior-mu)/sigma)-pnorm((inferior-mu)/sigma))

In [64]:
# Se calcula el estadístico de prueba
diferencias <- (values-esperadas)**2/esperadas
(chi_2 <- sum(diferencias))

[1] 2.359194

In [75]:
# Se calcula el p-value
pchisq(chi_2, length(values)-1, lower.tail = FALSE)

[1] 0.9927726

Con un p-value de 0.99 no se puede rechazar la hipótesis nula de que la muestra agrupada sigue una distribución normal. Al contrario, es casi completamente seguro afirmar que sí siga esa distribución, por el p-value tan alto.

### 12.3.5


__The following are the numbers of a particular organism found in 100 samples of water from a pond:__

0, 15 

1, 30 

2, 25 

3, 20 

4, 5

5, 4

6, 1

7, 0


Test the null hypothesis that these data were drawn from a Poisson distribution. Determine the p value for this test.

In [1]:
# Los valores y sus frecuencias
labels <- 0:7
values <- c(15, 30, 25, 20, 5, 4, 1, 0)

# El tamaño de la muestra
numero <- sum(values)

# Se usa que el promedio de la distribución de Poisson es igual al parámmetro
# de la distribución, lambda. Esto se hace para determinar la distribución de
# Poisson que se ajustará
(mean <- sum(labels*values/numero))

# Se calcula la distribución de frecuencias esperadas
esperadas <- numero*(exp(-mean)*(mean**labels)/factorial(labels))

[1] 1.86

In [79]:
# Se calcula el estadístico de prueba
diferencias <- (values-esperadas)**2/esperadas
(chi_2 <- sum(diferencias))

[1] 2.512458

In [80]:
# Se calcula el p-value
# En los grados de libertad se considera la restricción de que se calcula
# "a mano" el parámetro de la distribución
pchisq(chi_2, length(values)-2, lower.tail = FALSE)

[1] 0.8670708

Con un p-value de 0.87, no se puede rechazar la hipótesis nula de que la muestra agrupada sigue una distribución de Poisson con $\lambda=1.86$.

### 12.4.6


__The following table shows the results of a survey in which the subjects were a sample of 300 adults residing in a certain metropolitan area. Each subject was asked to indicate which of three policies they favored with respect to smoking in public places.__

Can one conclude from these data that, in the sampled population, there is a relationship between level of education and attitude toward smoking in public places? Let $\alpha = 0.05$.

In [111]:
# Se define la tabla de contingencia a estudiar
matriz <- t(matrix(c(5, 44, 23, 3, 75)))
matriz <- rbind(matriz, c(15, 100, 30, 5, 150))
matriz <- rbind(matriz, c(15, 40, 10, 10, 75))

# El tamaño de la matriz
dimension <- dim(matriz)

# Se calculan las frecuencias marginales
suma.columnas <- 1:dimension[2]
for (i in 1:dimension[2]){
  suma.columnas[i] <- sum(matriz[,i])
}

suma.filas <- 1:dimension[1]
for (i in 1:dimension[1]){
  suma.filas[i] <- sum(matriz[i,])
}

# El tamaño de la muestra
total <- sum(suma.filas)

# Se calculan las frecuencias esperadas
esperadas <- matrix(1, nrow = dimension[1], ncol = dimension[2])
for (i in 1:dimension[1]){
  for (j in 1:dimension[2]){
    esperadas[i,j] <- suma.filas[i]*suma.columnas[j]/total
  }
}

In [113]:
# Se calcula el estadístico de prueba
diferencias <- (matriz-esperadas)**2/esperadas
(chi_2 <- sum(diferencias))

[1] 22.50173

In [116]:
# Se calcula el p-value
pchisq(chi_2, (dimension[1]-1)*(dimension[2]-1), lower.tail = FALSE)

[1] 0.004066488

Con un p-value de 0.0041, con una significancia de 0.05, se puede rechazar la hipótesis nula de que no haya relación entre variables. Por lo tanto, el nivel educativo sí tiene una influencia en la percepción sobre fumar en espacios públicos.

### 12.5.5

__In a simple random sample of 250 industrial workers with cancer, researchers found that 102 had worked at jobs classified as “high exposure” with respect to suspected cancer-causing agents. Of the remainder, 84 had worked at “moderate exposure” jobs, and 64 had experienced no known exposure because of their jobs. In an independent simple random sample of 250 industrial workers from the same area who had no history of cancer, 31 worked in “high exposure” jobs, 60 worked in “moderate exposure” jobs, and 159 worked in jobs involving no known exposure to suspected cancercausing agents. Does it appear from these data that persons working in jobs that expose them to suspected cancer-causing agents have an increased risk of contracting cancer? Let $\alpha= 0.05$.__

In [117]:
# La tabla de contingencia
matriz <- t(matrix(c(102, 84, 64)))
matriz <- rbind(matriz, c(31, 60, 159))

# El tamaño de la matriz
dimension <- dim(matriz)

# Las frecuencias marginales
suma.columnas <- 1:dimension[2]
for (i in 1:dimension[2]){
  suma.columnas[i] <- sum(matriz[,i])
}

suma.filas <- 1:dimension[1]
for (i in 1:dimension[1]){
  suma.filas[i] <- sum(matriz[i,])
}

# El tamaño de la muestra
total <- sum(suma.filas)

# Se calculan las frecuencias esperadas
esperadas <- matrix(1, nrow = dimension[1], ncol = dimension[2])
for (i in 1:dimension[1]){
  for (j in 1:dimension[2]){
    esperadas[i,j] <- suma.filas[i]*suma.columnas[j]/total
  }
}

In [118]:
# El estadístico de prueba
diferencias <- (matriz-esperadas)**2/esperadas
(chi_2 <- sum(diferencias))

[1] 82.37311

In [119]:
# El p-value
pchisq(chi_2, (dimension[1]-1)*(dimension[2]-1), lower.tail = FALSE)

[1] 1.296901e-18

Con un p-value de $1.3\times10^{-18}$, para una significancia de 0.05, se puede rechazar la hipótesis nula de que no haya relación entre variables. Por lo tanto, el riesgo asociado a la exposición del trabajo en el que alguien se desempeña sí tiene influencia en su riesgo de contraer cáncer.

### 12.6.1

__The goal of a study by Tahmassebi and Curzon (A-13) was to determine if drooling in children with cerebral palsy is due to hypersalivation. One of the procedures toward that end was to examine the salivary buffering capacity of cerebral palsied children and controls. The following table gives the results.__

Test for a significant difference between cerebral palsied children and controls with respect to high or low buffering capacity. Let $\alpha=0.05$ and find the p value.

In [71]:
# Los datos
df <- data.frame(
  "palsy_no" = c(3, 7),
  "palsy_yes" = c(2, 8),
  row.names = c("medium", "high"),
  stringsAsFactors = FALSE
)

colnames(df) <- c("Control", "Palsy")

df <- t(df)

In [72]:
df

medium high
Control 3      7   
Palsy   2      8

Según la tabla J del libro (página A-57), como b=7, aunque la tabla no contiene explícitamente el valor, sí se sabe que el p-value si b=3, sería de p=0.035. Es decir, para b=7, se tiene un p-value de p>0.035. Por lo tanto, para una significancia del 0.05, se tiene que p>0.035>0.025. Es decir, no hay evidencia suficiente para rechazar la hipótesis nula de que hay diferencia entre el grupo de control y el de estudio.

In [75]:
# Para confirmar, se realiza una prueba de Fisher para verificar el resultado
fisher.test(df)


	Fisher's Exact Test for Count Data

data:  df
p-value = 1
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
  0.1441618 25.6043943
sample estimates:
odds ratio 
  1.668591 


Para un p-value similar a 1, se infiere que no se puede rechazar la hipótesis nula.

### 12.7.5

__According to Holben et al. (A-22), “Food insecurity implies a limited access to or availability of food or a limited/uncertain ability to acquire food in socially acceptable ways.” These researchers collected data on 297 families with a child in the Head Start nursery program in a rural area of Ohio near Appalachia. The main outcome variable of the study was household status relative to food security. Households that were not food secure are considered to be cases. The risk factor of interest was the absence of a garden from which a household was able to supplement its food supply. In the following table, the data are stratified by the head of household’s employment status outside the
home.__

Compute the Mantel–Haenszel common odds ratio with stratification by employment status. Use the Mantel–Haenszel chi-square test statistic to determine if we can conclude that there is an association between the risk factor and food insecurity. Let $\alpha= 0.05$.

In [7]:
# Los coeficientes del estrato 1 y 2
coeficientes.1 <- c(40, 37, 13, 38)
coeficientes.2 <- c(75, 38, 15, 33)

# El tamaño de la muestra 1 y 2
n.1 <- sum(coeficientes.1)
n.2 <- sum(coeficientes.2)

# Se calculan los coeficientes del estrato 1 y 2
e.1 <- (coeficientes.1[1]+coeficientes.1[2])*
       (coeficientes.1[1]+coeficientes.1[3])/n.1

e.2 <- (coeficientes.2[1]+coeficientes.2[2])*
       (coeficientes.2[1]+coeficientes.2[3])/n.2

v.1 <- (coeficientes.1[1]+coeficientes.1[2])*
       (coeficientes.1[3]+coeficientes.1[4])*
       (coeficientes.1[1]+coeficientes.1[3])*
       (coeficientes.1[2]+coeficientes.1[4])/((n.1**2)*(n.1-1))

v.2 <- (coeficientes.2[1]+coeficientes.2[2])*
       (coeficientes.2[3]+coeficientes.2[4])*
       (coeficientes.2[1]+coeficientes.2[3])*
       (coeficientes.2[2]+coeficientes.2[4])/((n.2**2)*(n.2-1))

In [6]:
# El estadístico de prueba y el p-value
(chi.2 <- (coeficientes.1[1]+coeficientes.2[1]-e.1-e.2)**2/(v.1+v.2))
pchisq(chi.2, 1, lower.tail = FALSE)

[1] 25.09515

[1] 5.457006e-07

Con un p-value de  $5.5\times10^{-7}$ , para una significancia de 0.05, se puede rechazar la hipótesis nula de que no haya relación entre variables. Por lo tanto, que la cabeza de familia trabaje fuera o dentro de casa afecta la proporción de infantes con seguridad alimenticia y que el hogar tenga un jardín o no.

In [8]:
# El odds ratio
(or <- (coeficientes.1[1]*coeficientes.1[4]/n.1+coeficientes.2[1]*coeficientes.2[4]/n.2)/
      (coeficientes.1[2]*coeficientes.1[3]/n.1+coeficientes.2[2]*coeficientes.2[3]/n.2))

[1] 3.733486

El odds ratio sugiere que los hogares sin hortaliza y sin sustentablidad alimenticia ocurren son 3.7 veces más prevalentes que en los hogares con hortaliza.